src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,130 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [initialMovie] = useState({
+    title: '',
+    description: '',
+    imgUrl: '',
+    imdbUrl: '',
+    imdbId: '',
+  });
+  const [movie, setMovie] = useState({ ...initialMovie });
+
+  const increment = () => {
+    setCount(prevCount => prevCount + 1);
+  };
+
+  const reset = () => {
+    setMovie(initialMovie);
+  };
+
+  const isNotValid =
+    movie.title.length === 0 ||
+    movie.imgUrl.length === 0 ||
+    movie.imdbUrl.length === 0 ||
+    movie.imdbId.length === 0;
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    onAdd(movie);
+    increment();
+    reset();
+
+    if (isNotValid) {
+      // eslint-disable-next-line no-useless-return
+      return;
+    }
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      method="POST"
+      className="NewMovie"
+      key={count}
+      onSubmit={handleSubmit}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={movie.title}
+        onChange={value =>
+          setMovie(prevState => ({
+            ...prevState,
+            title: value,
+          }))
+        }
         required
       />
 
-      <TextField name="description" label="Description" value="" />
+      <TextField
+        name="description"
+        label="Description"
+        value={movie.description}
+        onChange={value =>
+          setMovie(prevState => ({
+            ...prevState,
+            description: value,
+          }))
+        }
```

Comment:
so if you create `handleChange` as I mentioned before you will just pass this function for your `onChange` for each TextField like this
```suggestion
        onChange={handleChange}
```